app/models.py

```diff
@@ -1,3 +1,246 @@
+        try:
+            matching_movies = omdb.search_movie(title)
+        except:
+            raise ConnectionError
+
+        self.recentSearches.update({title:matching_movies})
+
+        #For now, only return most popular movie
+        highestIMDB = 0
+
+        if matching_movies:
+            movie = matching_movies.pop(0)
+
+            try:
+                movieObj = Movie.objects.get(imdbID=movie.imdb_id)
+            except Movie.DoesNotExist:
+                movieObj = None
+
+            if not movieObj:
+                response = omdb.request(i=movie.imdb_id, tomatoes=True, type='movie').json()
+                movieObj = Movie().fillWithJsonObject(response)
+
+            return movieObj
+        else:
+            return None
+
+class Tweet(models.Model):
+    text = models.CharField(max_length=256)
+    tweetID = models.BigIntegerField(unique=True)
+    created_at = models.CharField(max_length=256)
+    favorite_count = models.IntegerField()
+    lang = models.CharField(max_length=16)
+    location = models.CharField(max_length=256)
+    retweet_count = models.IntegerField()
+    user_name = models.CharField(max_length=256)
+    user_screen_name = models.CharField(max_length=256)
+    user_verified = models.BooleanField()
+    imdbID =  models.CharField(max_length=1024)
+    sentiment_score = models.FloatField(null=True, blank=True)
+
+    param_defaults = {
+        'text' : None,
+        'created_at' : None,
+        'favorite_count' : None,
+        'lang' : None,
+        'location' : None,
+        'retweet_count' : None,
+        'user_name' : None,
+        'user_screen_name' : None,
+        'user_verified' : False,
+        'sentiment_score' : False,
+        'imdbID': False,
+        'tweetID': False,
+    }
+
+    def fillWithStatusObject(self, tweet):
+        """
+        :param tweet: an object of the class twitter.Status (returnd by Twitter API)
+        :return updated_tweet: this tweet, updated with the data from the given tweet
+        """
+        if tweet is None or type(tweet) is not twitter.Status:
+            return self
+
+        # assume the Tweet is in the users location if we have no info
+        if type(tweet.location) is not str:
+            tweet.location = tweet.user.location
+        # assume the Tweet is in the users langauge if we have no info
+        if tweet.lang is None and tweet.user.lang is not None:
+            tweet.lang = tweet.user.lang
+
+        self.text=tweet.text
+        self.created_at=tweet.created_at
+        self.favorite_count=tweet.favorite_count
+        self.lang=tweet.lang
+        self.location=tweet.location
+        self.retweet_count=tweet.retweet_count
+        self.user_name=tweet.user.name
+        self.user_screen_name=tweet.user.screen_name
+        self.user_verified=tweet.user.verified
+        self.tweetID = tweet.id
+        self.imdbID = tweet.imdbID
+
+        self.assignSentimentScore()
+
+        try:
+            self.save()
```

Comment:
We might also want to add more checks for each field in the ```Tweet``` model before trying to save to minimize the chance that the database rejects the ```save()```.